crowbar_framework/app/assets/javascripts/jquery/nodeList.js

```diff
@@ -417,29 +417,37 @@
       return [0, 0, 0];
     };
 
-    var cmp = function(op, version1, version2) {
-      var as_version1 = to_version(version1);
-      var as_version2 = to_version(version2);
-      var result = as_version1.map(function(v1, i) {
-        switch (op) {
-          case ">=":
-            return v1 >= as_version2[i];
-          case ">":
-            return v1 > as_version2[i];
-          case "<=":
-            return v1 <= as_version2[i];
-          case "<":
-            return v1 < as_version2[i];
-          case "==":
-            return v1 == as_version2[i];
-          default:
-            return false;
+    var cmp = function(version1, version2) {
+      var ans = 0;
+      for (var i=0; i<version1.length; i++) {
+        if (version1[i] < version2[i]) {
```

Comment:
There's a (correct, iirc) assumption to at least document, or to remove from the code, that `version1.length == version2.length`.